cachi2/core/package_managers/yarn/main.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from cachi2.core.models.input import Request
+from cachi2.core.models.output import Component, EnvironmentVariable, RequestOutput
+from cachi2.core.package_managers.yarn.project import Project, YarnRc
+from cachi2.core.package_managers.yarn.resolver import (
+    create_component_from_package,
+    resolve_dependency_info,
+)
+from cachi2.core.package_managers.yarn.utils import run_yarn_cmd
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+
+
+def fetch_yarn_source(request: Request) -> RequestOutput:
+    """Process all the yarn source directories in a request."""
+    components = []
+
+    for package in request.yarn_packages:
+        path = request.source_dir.join_within_root(package.path)
+        project = Project.from_source_dir(path)
+        components.extend(_resolve_yarn_project(project, request.output_dir))
+
+    return RequestOutput.from_obj_list(
+        components, _generate_environment_variables(), project_files=[]
+    )
+
+
+def _resolve_yarn_project(project: Project, output_dir: RootedPath) -> list[Component]:
+    """Process a request for a single yarn source directory.
+
+    :param project: the directory to be processed.
+    :param output_dir: the directory where the prefetched dependencies will be placed.
+    """
+    _configure_yarn_version(project)
+
+    try:
+        _prepare_for_prefetch(project.yarn_rc, output_dir)
+        packages = resolve_dependency_info(project.source_dir)
+        _fetch_dependencies(project)
+    except Exception:
+        _restore_files(project)
+        raise
+
+    _restore_files(project)
+
+    return [create_component_from_package(package, project.yarn_rc) for package in packages]
+
+
+def _configure_yarn_version(project: Project) -> None:
+    """Resolve the yarn version and set it in the package.json file if needed.
+
+    :raises PackageRejected: in case the yarn version can't be determined, or if
+    there is a mismatch between the version in package.json and yarnrc.yml.
+    """
+    pass
+
+
+def _prepare_for_prefetch(yarnrc: YarnRc, output_dir: RootedPath) -> None:
+    """Change the configuration in yarnrc to perform the prefetch.
+
+    :param output_dir: the directory where the prefetched dependencies will be placed.
+    """
+    pass
+
+
+def _fetch_dependencies(project: Project) -> None:
+    """Fetch dependencies using 'yarn install'.
+
+    :raises SubprocessCallError: if the 'yarn install' command fails.
+    """
+    run_yarn_cmd(["yarn", "--help"], project.source_dir)
+    pass
```

Comment:
Maybe call out that for zero-installs, we're not fetching but verifying (although verifying does involve fetching because of `--check-cache`)